### DISCUSSION TITLE
Possible to use vite-plugin-ssr in a monorepo? (NX)

### DISCUSSION THREAD

**Discussion by simonwjackson** [vikejs/vike#685] [Discussion]
Is it possible to move pages outside of the root of the application?

This is what I would like to do:

```
app/my-app/src/pages       <--- Move from here..
libs/my-app/pages          <--- ..to here
```

For context, I am working in a monorepo with a handful of web apps `(./apps)` that share components. We use a domain driven approach to separate our concerns into small libraries `(./libs)`.

However, after scanning the docs I'm not sure something like this would be possible today. Any thoughts on how this could be achieved?

**Comment by brillout** [comment]
Either move `vite.config.js` (or set [`config.root`](https://vitejs.dev/config/shared-options.html#root)) to a directory that is a common ancestor, or use a VPS extension (https://github.com/brillout/docpress/blob/581bdebb76bd9bb47f401f7879eb67f5e796f0e6/src/vite.config.ts#L24).

Note that extensions are experimental and subject to change. But let me know if your company needs this sooner rather than later.

**Reply by DreamwareDevelopment** [reply]
I'm trying to do something similar, I'd like to keep the code that runs on the server completely separate from client code. I have an NX repo with the client code in `client` and the server in `frontend-server`.  I'd like to have two `pages` directories. One for ssr and the other for spa. From what I can tell about the extension is that maybe that'll work for me. Also, is it okay to have multiple _default.page.client.tsx files, one for each of these directories?

**Reply by brillout** [reply]
Check out https://vite-plugin-ssr.com/file-structure#domain-driven. Your `pages/` will still share the same `vite.config.js` though.

Yes, you can have multilple `renderer/` directories (and therefore mutliple `_default.*`), e.g. one for each `pages/`.

Implementing several `renderer/` requires a non-negligle amount of manual work on your part. If you don't want to do that manual work, then check out [vike-react](https://github.com/vikejs/vike-react) which features an `ssr: false` flag (see example) to opt-out of SSR. Vike-react is in beta but it's fairly stable already and issues are fixed promptly.

**Reply by NiklasBuchfink** [reply]
> I'm trying to do something similar, I'd like to keep the code that runs on the server completely separate from client code. I have an NX repo with the client code in `client` and the server in `frontend-server`. I'd like to have two `pages` directories. One for ssr and the other for spa.

We have a similar setup at inlang now and it seems the two vite-plugin-ssr configurations are conflicting with each other:
1. Only the page directory of the first imported configuration will be used for routing. The other is ignored.
2. It always uses the same renderers
This results in a check whether the route exists in page directory 1 and page directory 2 is rendered.

@brillout Is there a way that route `/editor` is handled by one vite-plugin-ssr setup and the other routes by the second setup when both are added to the same express server router?

**Reply by brillout** [reply]
@NiklasBuchfink I don't understand your use case. Elaborate in a new GitHub Discussion (seems be a completely different question than OP's) while focusing more on describing what you're trying to achieve (why do you have two Vike setups to begin with?) and slightly less about your attempts to get there.
